dhcpv6/dhcpv6relay.go

```diff
@@ -10,125 +10,94 @@ import (
 	)
 	return ret
 }
 
 // ToBytes returns the serialized version of this relay message as defined by
-// RFC 3315, Section 6.
-func (r *DHCPv6Relay) ToBytes() []byte {
+// RFC 3315, Section 7.
+func (r *RelayMessage) ToBytes() []byte {
 	buf := uio.NewBigEndianBuffer(make([]byte, 0, RelayHeaderSize))
-	buf.Write8(byte(r.messageType))
-	buf.Write8(byte(r.hopCount))
-	buf.WriteBytes(r.linkAddr.To16())
-	buf.WriteBytes(r.peerAddr.To16())
-	buf.WriteBytes(r.options.ToBytes())
+	buf.Write8(byte(r.MessageType))
+	buf.Write8(byte(r.HopCount))
+	buf.WriteBytes(r.LinkAddr.To16())
+	buf.WriteBytes(r.PeerAddr.To16())
+	buf.WriteBytes(r.Options.ToBytes())
 	return buf.Data()
 }
 
-func (r *DHCPv6Relay) SetMessageType(messageType MessageType) {
-	// not enforcing if message type is not a RELAY_FORW or a RELAY_REPL message
-	r.messageType = messageType
-}
-
-func (r *DHCPv6Relay) HopCount() uint8 {
-	return r.hopCount
-}
-
-func (r *DHCPv6Relay) SetHopCount(hopCount uint8) {
-	r.hopCount = hopCount
-}
-
-func (r *DHCPv6Relay) LinkAddr() net.IP {
-	return r.linkAddr
-}
-
-func (r *DHCPv6Relay) SetLinkAddr(linkAddr net.IP) {
-	r.linkAddr = linkAddr
-}
-
-func (r *DHCPv6Relay) PeerAddr() net.IP {
-	return r.peerAddr
-}
-
-func (r *DHCPv6Relay) SetPeerAddr(peerAddr net.IP) {
-	r.peerAddr = peerAddr
+// GetOption returns the options associated with the code.
+func (r *RelayMessage) GetOption(code OptionCode) []Option {
+	return r.Options.Get(code)
 }
 
-func (r *DHCPv6Relay) Options() []Option {
-	return r.options
+// GetOneOption returns the first associated option with the code from this
+// message.
+func (r *RelayMessage) GetOneOption(code OptionCode) Option {
+	return r.Options.GetOne(code)
 }
 
-func (r *DHCPv6Relay) GetOption(code OptionCode) []Option {
-	return r.options.Get(code)
-}
-
-func (r *DHCPv6Relay) GetOneOption(code OptionCode) Option {
-	return r.options.GetOne(code)
-}
-
-func (r *DHCPv6Relay) SetOptions(options []Option) {
-	r.options = options
-}
-
-func (r *DHCPv6Relay) AddOption(option Option) {
-	r.options.Add(option)
+// AddOption adds an option to this message.
+func (r *RelayMessage) AddOption(option Option) {
+	r.Options.Add(option)
 }
 
 // UpdateOption replaces the first option of the same type as the specified one.
-func (r *DHCPv6Relay) UpdateOption(option Option) {
-	r.options.Update(option)
+func (r *RelayMessage) UpdateOption(option Option) {
+	r.Options.Update(option)
 }
 
-func (r *DHCPv6Relay) IsRelay() bool {
+// IsRelay returns whether this is a relay message or not.
+func (r *RelayMessage) IsRelay() bool {
 	return true
 }
 
-// Recurse into a relay message and extract and return the inner DHCPv6Message.
-// Return nil if none found (e.g. not a relay message).
-func (d *DHCPv6Relay) GetInnerMessage() (DHCPv6, error) {
+// GetInnerMessage recurses into a relay message and extract and return the
+// inner Message.  Return nil if none found (e.g. not a relay message).
+func (r *RelayMessage) GetInnerMessage() (DHCPv6, error) {
```

Comment:
In order to make it consistent with the rest of the changes, should this return a *Message instead of an interface?

Reply:
done